201-representing-CNAB-in-OCI.md

```diff
@@ -0,0 +1,132 @@
+
+```json
+{
+  "schemaVersion": "v1.0.0",
+  "name": "helloworld",
+  "version": "0.1.1",
+  "description": "A short description of your bundle",
+  "keywords": ["helloworld", "cnab", "tutorial"],
+  "maintainers": [
+    {
+      "name": "Jane Doe",
+      "email": "jane.doe@example.com",
+      "url": "https://example.com"
+    }
+  ],
+  "invocationImages": [
+    {
+      "imageType": "docker",
+      "image": "cnab/helloworld:0.1.1",
+      "size": 942,
+      "contentDigest": "sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6",
+      "mediaType": "application/vnd.docker.distribution.manifest.v2+json"
+    }
+  ],
+  "images": null,
+  "parameters": null,
+  "credentials": null
+}
+```
+
+The `bundle.json` is serialized into canonical JSON and is pushed as a blob to the registry. This blob is then referenced by its digest (for example `sha256:e91b9dfcbbb3b88bac94726f276b89de46e4460b55f6e6d6f876e666b150ec5b`) as the object config.
+
+The bundle is encapsulated in an OCI manifest which is then referenced as part of an OCI index (or manifest list):
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.cnab.bundle.config.v1+json",
+    "digest": "sha256:e91b9dfcbbb3b88bac94726f276b89de46e4460b55f6e6d6f876e666b150ec5b",
+    "size": 498
+  },
+  "layers": null
+}
+```
+
+The bundle file MUST be stored in the OCI registry, as it is the canonical representation of a CNAB bundle, and is the object that is used to compute the signature for a thin bundle.
+While implementations MAY choose to surface information from the bundle as top-level annotations in the OCI representations, they MUST store the unmodified bundle file as a blob.
+
+The bundle file and registry location MAY be used by implementations to generate relocation maps at runtime.
+
+### Invocation images
+
+> In practice most images are built as Docker v2 images so we should allow these
+
+Each invocation image SHOULD be packaged as an OCI image but MAY be packaged as a Docker v2 image. If there are multiple invocation images, these SHOULD be referenced by an OCI index but MAY be referenced by a Docker manifest list.
+
+### Component images
+
+> In practice most images are built as Docker v2 images so we should allow these
+
+Each component image SHOULD be packaged as an OCI image but MAY be packaged as a Docker v2 image. In the case that a component has multiple equivalent forms, such as a multiarch container image, each component MUST be packaged and then SHOULD be referenced by an OCI index but MAY be referenced by a Docker manifest list.
+
+## Top-level representation
+
+The top-level representation of the CNAB SHOULD be an OCI index but may be a Docker manifest list.
+
+The `manifests` list MUST be filled in the following order:
+
+- A reference to packaged `bundle.json` manifest
+- References to the invocation images
+- References to the component images
+
+### Example
```

Comment:
Hmm, sorry, I'm confused. The example doesn't seem to match the preceding prescription, but I might just be missing something. Could someone please explain?

Reply:
This is happening because the example itself only contains the reference to `bundle.json` and an invocation image, but it does follow this structure.